apps/spark/src/main/java/com/linkedin/openhouse/jobs/spark/Operations.java

```diff
@@ -216,14 +216,13 @@ public void expireSnapshots(String fqtn, int maxAge, String granularity, int ver
   public void expireSnapshots(Table table, int maxAge, String granularity, int versions) {
     ExpireSnapshots expireSnapshotsCommand = table.expireSnapshots().cleanExpiredFiles(false);
 
-    // maxAge is always defined with granularity
-    if (!granularity.isEmpty()) {
-      TimeUnit timeUnitGranularity = TimeUnit.valueOf(granularity.toUpperCase());
-      long expireBeforeTimestampMs =
-          System.currentTimeMillis() - timeUnitGranularity.toMillis(maxAge);
-      log.info("Expiring snapshots for table: {} older than {}ms", table, expireBeforeTimestampMs);
-      expireSnapshotsCommand.expireOlderThan(expireBeforeTimestampMs).commit();
-    }
+    // maxAge will always be defined
```

Comment:
how do we make sure this is backward compatible, is there some coordination required for the deployment so that requests received on jobs-service will always have maxAge ?

Reply:
It's already done previously, since the scheduler reads the config https://github.com/linkedin/openhouse/blob/7f66a730ec7d611335917e29bfed01050090c397/apps/spark/src/main/java/com/linkedin/openhouse/jobs/scheduler/tasks/TableSnapshotsExpirationTask.java#L38 it will read the table properties and set accordingly. Since the change has not been rolled out yet and no tables currently use this property there's nothing to coordinate here, as we always fallback to maxAge = 3 days.